meta-balena-common/classes/kernel-balena.bbclass

```diff
@@ -63,6 +63,33 @@ inherit kernel-resin-noimage
 
 BALENA_DEFCONFIG_NAME ?= "resin-defconfig"
 
+def get_kernel_version(d):
```

Comment:
It would be kind of nice if, in addition to checking by kernel version, we could tie into Kconfig, and enable configs conditionally based on the availability or presence of configs. For example, instead of tracking down the version a feature/config was introduced in, we could have a function that checks for the availability or status of `CONFIG_FW_LOADER_COMPRESS`.  We could then enables it and compress the firmware.

It looks like there's a Python parser we could look into at some point. https://github.com/ulfalizer/Kconfiglib

Reply:
Not sure I understand how that would work. Given that we are dynamically configuring the kernel, we could check for a setting that is not introduced until later, in other words, we would not be checking against the final config. I think kernel versions are more clear.